netty/src/main/java/io/grpc/netty/Utils.java

```diff
@@ -89,6 +89,10 @@ class Utils {
       = new DefaultEventLoopGroupResource(1, "grpc-nio-boss-ELG", EventLoopGroupType.NIO);
   public static final Resource<EventLoopGroup> NIO_WORKER_EVENT_LOOP_GROUP
       = new DefaultEventLoopGroupResource(0, "grpc-nio-worker-ELG", EventLoopGroupType.NIO);
+  public static final Resource<EventLoopGroup> UDS_CHANNELS_EVENT_LOOP_GROUP =
+      isEpollAvailable() ? new DefaultEventLoopGroupResource(1, "UdsChannels",
```

Comment:
We don't need a new event loop, right? We can use the one assigned to DEFAULT_WORKER_EVENT_LOOP_GROUP when epoll is available, right?

Reply:
How do I reconcile this with the [comment above](https://github.com/grpc/grpc-java/pull/9113#discussion_r862200474)?

If I remove `UDS_CHANNELS_EVENT_LOOP_GROUP` and use `DEFAULT_WORKER_EVENT_LOOP_GROUP` then I don't get the check of `UDS_CHANNELS_EVENT_LOOP_GROUP != null` to indirectly test `Epoll.isAvailable()` 

May be use `EPOLL_EVENT_LOOP_GROUP_CONSTRUCTOR != null` instead as a check?

Reply:
This is pointing to the allocation, not the variable. I was mostly leaving it up to you to which approach. Ideas:
1. Initialize UDS_CHANNELS_EVENT_LOOP_GROUP to DEFAULT_WORKER_EVENT_LOOP_GROUP or `null` in the `static {}`
2. Key off of EPOLL_EVENT_LOOP_GROUP_CONSTRUCTOR (seems a bit hacky, because what keeps this in sync with the event loop field?)
3. Have a `boolean USING_EPOLL_BY_DEFAULT`
4. Expose `Utils.isEpollAvailable()` and call it from the name resolver (seems a bit hacky, because what keeps this in sync with the event loop field?)

I probably would have approved the ones I'm labeling hacky here.

Reply:
So I am using `Utils.EPOLL_DOMAIN_CLIENT_CHANNEL_TYPE == null`   since that too indirectly checks isEpollAvailable. I am also getting rid of UDS_CHANNELS_EVENT_LOOP_GROUP and directly using DEFAULT_WORKER_EVENT_LOOP_GROUP.